internal/builders/archive/src/index.ts

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2024 SLSA Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WIHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as github from "@actions/github";
+import * as core from "@actions/core";
+import * as inputs from "./inputs";
+import * as attestation from "./attestation";
+import * as crypto from "crypto";
+import * as tscommon from "tscommon";
+
+async function run(): Promise<void> {
+  /* Test locally. Requires a GitHub token:
+    $ env INPUT_SLSA-WORKFLOW-INPUTS='{"formats":"zip tar.gz"}' \
+    GITHUB_REPOSITORY="slsa-framework/slsa-policy" \
+    GITHUB_REF_NAME="v0.0.1" \
+    SLSA_OUTPUTS_ARTIFACTS_FILE="output.json" \
+    GITHUB_TOKEN="$(echo $GH_TOKEN)" \
+    nodejs ./dist/index.js
+  */
+  try {
+    // Get token.
+    const ghToken = inputs.getEnvVariable("GITHUB_TOKEN");
+    // Get workflow inputs.
+    const workflowInputs = inputs.getActionInput("slsa-workflow-inputs");
+    const jsonInputs = JSON.parse(workflowInputs);
+    const f = jsonInputs["formats"];
+    if (!f) {
+      throw new Error("formats empty");
+    }
+    const formats = inputs.parseFormats(f);
+    const apiFormats = inputs.formatsToAPI(formats);
+
+    // Verify that upload-assets is set to true.
+    const uploadAssets = jsonInputs["upload-assets"].trim();
+    if (uploadAssets !== "true") {
+      throw new Error(`expected upload-assets to be true, got ${uploadAssets}`);
+    }
```

Comment:
I'm guessing you mean for this to intentionally fail, but perhaps towards the end of the workflow